tests/unit-tests/dds/DCPS/RTPS/RtpsCoreTypeSupportImpl.cpp

```diff
@@ -50,11 +52,59 @@ void extract_Parameter_user_data(const std::string& user_data, const Encoding en
   }
 }
 
+// ---------- ---------- ---------- ---------- ----------
+template <typename Seq, typename Element>
+bool insert_sequence(Serializer& s, const std::vector<Element>& v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  Seq seq;
+  seq.length(length);
+  for (CORBA::ULong i = 0; i < length; ++i) {
+    seq[i] = v[i];
+  }
+  return s << seq;
+}
+
+template <typename Seq, typename Element>
+void extract_sequence(Serializer& s, const std::vector<Element>& v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  Seq seq;
+  ASSERT_TRUE(s >> seq);
+  ASSERT_EQ(seq.length(), length);
+  for (CORBA::ULong i = 0; i < length; ++i) {
+    EXPECT_EQ(seq[i], v[i]);
+  }
+}
+
+template <typename Seq>
+bool extract_sequence_with_unverified_length(const CORBA::ULong length = 0)
+{
+  const CORBA::ULong data[] = {length, 1000};
+  ACE_Message_Block mb(512);
+  mb.copy((const char*)data, sizeof(data));
+  Serializer s(&mb, Encoding(Encoding::KIND_XCDR1));
+
+  Seq seq;
+  return s >> seq;
+}
+
+// ---------- ---------- ---------- ---------- ----------
+void extract_FilterResult_t(const std::vector<CORBA::Long>& v)
+{
+  const CORBA::ULong length = static_cast<CORBA::ULong>(v.size());
+  ACE_Message_Block mb(length * sizeof(CORBA::Long) + 128);
```

Comment:
You shouldn't use `sizeof` for any CDR buffer size, because `sizeof` is for C++ objects, not CDR ones. Please use `serialized_size`.